chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,323 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+        ]
+
+        # Ensure we have an xch coin of high enough amount
+        assert isinstance(fee, uint64)
+        total_amount = len(metadata_list) + fee
+        if xch_coins is None:
+            xch_coins = await self.standard_wallet.select_coins(uint64(total_amount))
+        assert len(xch_coins) > 0
+
+        # set the chunk size for the spend bundle we're going to create
+        chunk_size = len(metadata_list)
+
+        # Because bulk minting may not mint all the NFTs in one bundle, we
+        # calculate the edition numbers that will be used in the intermediate
+        # puzzle based on the starting edition number given, and the size of the
+        # chunk going into this spend bundle
+        mint_number_end = mint_number_start + chunk_size
+
+        # Empty set to load with the announcements we will assert from DID to
+        # match the announcements from the intermediate launcher puzzle
+        did_announcements: Set = set()
+        puzzle_assertions: Set = set()
+        amount = uint64(1)
+        intermediate_coin_spends = []
+        launcher_spends = []
+        launcher_ids = []
+        eve_spends = []
+        transfer_spends = []
+        p2_inner_puzzle = await self.standard_wallet.get_new_puzzle()
+
+        # Loop to create each intermediate coin, launcher, eve and (optional) transfer spends
+        for mint_number in range(mint_number_start, mint_number_end):
+            # Create  the puzzle, solution and coin spend for the intermediate launcher
+            intermediate_launcher_puz = did_wallet_puzzles.DID_NFT_LAUNCHER_MOD.curry(
+                did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, mint_number, mint_total
+            )
+            primaries.append(
+                AmountWithPuzzlehash(
+                    {
+                        "puzzlehash": intermediate_launcher_puz.get_tree_hash(),
+                        "amount": uint64(0),
+                        "memos": [intermediate_launcher_puz],
+                    }
+                )
+            )
+            intermediate_launcher_sol = Program.to([did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, mint_number, chunk_size])
+            intermediate_launcher_coin = Coin(did_coin.name(), intermediate_launcher_puz.get_tree_hash(), uint64(0))
+            intermediate_launcher_coin_spend = CoinSpend(
+                intermediate_launcher_coin, intermediate_launcher_puz, intermediate_launcher_sol
+            )
+            intermediate_coin_spends.append(intermediate_launcher_coin_spend)
+
+            # create an ASSERT_COIN_ANNOUNCEMENT for the DID spend. The
+            # intermediate launcher coin issues a CREATE_COIN_ANNOUNCEMENT of
+            # the mint_number and mint_total for the launcher coin it creates
+            intermediate_announcement_message = std_hash(int_to_bytes(mint_number) + int_to_bytes(mint_total))
+            did_announcements.add(std_hash(intermediate_launcher_coin.name() + intermediate_announcement_message))
```

Comment:
I don't think we need any data, just the coin ID implicit in `ASSERT_COIN_ANNOUNCEMENT` is fine.  It doesn't matter so much because it doesn't really result in any extra data on chain but it does make `DID_NFT_LAUNCHER_MOD` a bit smaller

Reply:
The values are there to make sure that each intermediate launcher coin is unique. If we curry them in and don't use them for anything I'm not sure whether the compiled clvm just drops the values so we end up with non-unique launchers. Have left as is for now, happy to remove if this would work and be preferred though

Reply:
Currying changes the puzzle hash whether you use the values or not, but like I said, it doesn't really matter so I'm happy to leave it as it is.